File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetssupplier.go
package shared

import (
	"time"
)

type CodatDataContractsDatasetsSupplier struct {
	Addresses          []CodatDataContractsDatasetsAddress                       `json:"addresses,omitempty"`
	ContactName        *string                                                   `json:"contactName,omitempty"`
	DefaultCurrency    *string                                                   `json:"defaultCurrency,omitempty"`
	EmailAddress       *string                                                   `json:"emailAddress,omitempty"`
	ID                 *string                                                   `json:"id,omitempty"`
	Metadata           *CodatDataContractsDatasetsMetadata                       `json:"metadata,omitempty"`
	ModifiedDate       *time.Time                                                `json:"modifiedDate,omitempty"`
	Phone              *string                                                   `json:"phone,omitempty"`
	RegistrationNumber *string                                                   `json:"registrationNumber,omitempty"`
	SourceModifiedDate *time.Time                                                `json:"sourceModifiedDate,omitempty"`
	Status             CodatDataContractsDatasetsSupplierStatusEnum              `json:"status"`
	SupplementalData   *CodatDataContractsDatasetsDataInterfacesSupplementalData `json:"supplementalData,omitempty"`
	SupplierName       *string                                                   `json:"supplierName,omitempty"`
	TaxNumber          *string                                                   `json:"taxNumber,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetssupplier.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsAddress } from "./codatdatacontractsdatasetsaddress";
import { CodatDataContractsDatasetsDataInterfacesSupplementalData } from "./codatdatacontractsdatasetsdatainterfacessupplementaldata";
import { CodatDataContractsDatasetsMetadata } from "./codatdatacontractsdatasetsmetadata";
import { CodatDataContractsDatasetsSupplierStatusEnum } from "./codatdatacontractsdatasetssupplierstatusenum";
import { Expose, Transform, Type } from "class-transformer";


export class CodatDataContractsDatasetsSupplier extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsAddress })
  @Expose({ name: "addresses" })
  @Type(() => CodatDataContractsDatasetsAddress)
  addresses?: CodatDataContractsDatasetsAddress[];

  @SpeakeasyMetadata()
  @Expose({ name: "contactName" })
  contactName?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "defaultCurrency" })
  defaultCurrency?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "emailAddress" })
  emailAddress?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "id" })
  id?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "metadata" })
  @Type(() => CodatDataContractsDatasetsMetadata)
  metadata?: CodatDataContractsDatasetsMetadata;

  @SpeakeasyMetadata()
  @Expose({ name: "modifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  modifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "phone" })
  phone?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "registrationNumber" })
  registrationNumber?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "sourceModifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  sourceModifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status: CodatDataContractsDatasetsSupplierStatusEnum;

  @SpeakeasyMetadata()
  @Expose({ name: "supplementalData" })
  @Type(() => CodatDataContractsDatasetsDataInterfacesSupplementalData)
  supplementalData?: CodatDataContractsDatasetsDataInterfacesSupplementalData;

  @SpeakeasyMetadata()
  @Expose({ name: "supplierName" })
  supplierName?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "taxNumber" })
  taxNumber?: string;
}